server/http_tunnel.go

```diff
@@ -31,22 +32,36 @@ func StartHTTPSTunnel(addr, externalProxy, certFile, KeyFile string) {
 		}}
 		proxy.ConnectDial = proxy.NewConnectDialToProxy(externalProxy)
 	}
+
+	proxy.OnRequest().HandleConnectFunc(func(host string, ctx *goproxy.ProxyCtx) (*goproxy.ConnectAction, string) {
+		log.Printf("HTTP Tunnel: got CONNECT request %s\n", host)
+		return nil, host
+	})
+
 	proxy.OnRequest().DoFunc(
 		func(r *http.Request, ctx *goproxy.ProxyCtx) (*http.Request, *http.Response) {
 			// Leave the request untouched.  Log a message for debugging purposes.
 			log.Printf("HTTP Tunnel: got request %s\n", r.URL)
 			return r, nil
 		})
 
-	if certFile == "" || KeyFile == "" {
+	if username != "" || password != "" {
```

Comment:
I am kind of curious here. Is it okay to let one of this to be empty? If both of them are not empty, and then we are going to perform the basic authentication?

Reply:
I'm using empty string to mean that no username or password is set.  In command line parsing, we're not giving the user a way to set empty string to be the username or password.  Therefore, by the time we get here, if both are empty, that means neither command line options were specified.

Reply:
It is possible to let one of them be empty.

Reply:
It's possible for the password to be empty, but it doesn't make sense for the username to be empty if a password was specified.  How is that case handled?

Reply:
It's possible to form a valid basic auth header with an empty username or an empty password, or both.  But since it's confusing, I'll require both to be set on the command line.

Reply:
Nah, that's ok.  If it's actually a valid scenario where a proxy is configured to have an empty username but a nonempty password, then we should allow it.